agenda-services/src/main/java/org/exoplatform/agenda/service/AgendaEventGuestServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package org.exoplatform.agenda.service;
+
+import org.exoplatform.agenda.model.*;
+import org.exoplatform.agenda.storage.AgendaEventGuestStorage;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AgendaEventGuestServiceImpl implements AgendaEventGuestService{
+
+    private AgendaEventGuestStorage   agendaEventGuestStorage;
+
+    public AgendaEventGuestServiceImpl(AgendaEventGuestStorage agendaEventGuestStorage) {
+        this.agendaEventGuestStorage = agendaEventGuestStorage;
+    }
+
+    @Override
+    public List<GuestUser> getEventGuests(long eventId) {
+        return agendaEventGuestStorage.getEventGuests(eventId);
+    }
+
+    @Override
+    public void saveEventGuests(long eventId, List<GuestUser> guests) {
+        List<GuestUser> oldGuests = getEventGuests(eventId);
+        List<GuestUser> newGuests = guests == null ? Collections.emptyList() : guests;
+        processGuestsToDelete(oldGuests, newGuests, eventId);
+        processGuestsToCreate(oldGuests, newGuests, eventId);
+    }
+
+    @Override
+    public void deleteEventGuests(long eventId) {
+       agendaEventGuestStorage.deleteEventGuests(eventId);
+    }
+
+    public  void processGuestsToCreate(List<GuestUser> oldGuests,List<GuestUser> newGuests,Long eventId){
+        List<GuestUser> guestsToCreate =
+                newGuests.stream()
+                        .filter(guest -> oldGuests.stream()
```

Comment:
Could you do something like this instead having to stream imbricated ?
```suggestion
                        .filter(guest -> oldGuests.contains(guest)
```